### PR TITLE
Add project Cppcheck

### PR DESCRIPTION
**Pull request by danmar**
I am the maintainer for Cppcheck and would like to improve the fuzz testing of this project.

### CODE REVIEW

**Comment by danmar** [comment]
hmm.. I think we better wait with oss-fuzz until we are more stable. I'll open a new PR if that happens..

**Comment by inferno-chromium** [comment]
@danmar - being stable is not a requirement for oss-fuzz integration. in fact, you will get better reports with stacktraces, etc making it easier to

**Comment by danmar** [comment]
thanks for pointing out `check_build` I had missed that step. I have fixed it now.

> Sorry for late response. We will get this in first thing monday ?

No problem.

File: projects/cppcheck/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER daniel.marjamaki@gmail.com
+
+RUN git clone https://github.com/orbitcowboy/afl_cppcheck.git afl_cppcheck
```
Comment (inferno-chromium) [inline projects/cppcheck/Dockerfile]: nit: is there any plan to move code from https://github.com/orbitcowboy/afl_cppcheck.git into the main dev repo https://github.com/danmar/cppcheck.git somepoint in future. is @orbitcowboy a cppcheck dev ?

File: projects/cppcheck/build.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# limitations under the License.
+#
+################################################################################
+
+# build fuzzer
+
+cd fuzzer-cli
+make oss-fuzz-client
+cp oss-fuzz-client $OUT/
+
```
Comment (inferno-chromium) [inline projects/cppcheck/build.sh]: at some point, you should add a seed corpus and dictionary for fuzzing efficiency.

File: projects/cppcheck/Dockerfile (hunk 1 above)
Comment (danmar) [reply projects/cppcheck/Dockerfile]: I do not have such a plan. Yes, orbitcowboy is an active cppcheck dev.

File: projects/cppcheck/build.sh (hunk 2 above)
Comment (danmar) [reply projects/cppcheck/build.sh]: For information..

the client takes the binary input from the fuzzer and conerts the ones and zeros it get into code.

For instance imagine that the fuzzer generates 2 bytes "AA". That is translated into this code:

    void f() {
        x=( ){ } >= %;
    }

"inferno" is translated to:

    void f() {
        x= return int enum int typedef && return continue >= ^ <;
    }

It's not 100% valid but we can tweak that...

Comment (inferno-chromium) [reply projects/cppcheck/build.sh]: Sounds good, thanks!

File: projects/cppcheck/Dockerfile (hunk 1 above)
Comment (inferno-chromium) [reply projects/cppcheck/Dockerfile]: No need then since orbitcowboy@ is active dev.

**Pull request merged**